test/SILOptimizer/licm_multiend.sil

```diff
@@ -0,0 +1,121 @@
+// RUN: %target-sil-opt -assume-parsing-unqualified-ownership-sil -enable-sil-verify-all %s -licm | %FileCheck %s
+
+sil_stage canonical
+
+import Builtin
+import Swift
+import SwiftShims
+
+var x: Int
+
+let reversedArray: ReversedCollection<[Int]>
+
+// x
+sil_global hidden @$S3tmp1xSivp : $Int
+
+// reversedArray
+sil_global hidden [let] @$S3tmp13reversedArrays18ReversedCollectionVySaySiGGvp : $ReversedCollection<Array<Int>>
+
+// _swiftEmptyArrayStorage
+sil_global @_swiftEmptyArrayStorage : $_SwiftEmptyArrayStorage
+
+
+// CHECK-LABEL: sil hidden @multi_end_licm : $@convention(thin) () -> () {
+// CHECK: bb2:
+// CHECK: [[GLOBALVAR:%.*]] = global_addr @$S3tmp1xSivp : $*Int
+// CHECK: [[BEGINA:%.*]] = begin_access [modify] [dynamic] [no_nested_conflict] [[GLOBALVAR]] : $*Int
+// CHECK-NEXT: br bb4({{.*}} : $Builtin.Int64)
+// CHECK: bb4({{.*}} : $Builtin.Int64)
+// CHECK: cond_br {{.*}}, bb5, bb7
```

Comment:
Please use a symbolic REGEX for the checked block names. Otherwise it's confusing how they relate to the original blocks. In the original code you could use the same symbol for the block to be really clear.